.github/CI/spack_setup.sh

```diff
@@ -0,0 +1,56 @@
+# This file should be "sourced" into your environment
+# to set up the spack repository
+
+DIR=${HOME}/spack
+
+#
+# Only update the spack clone once a day
+#
+echo "::group::Spack setup"
+if [ -r ${DIR}/.git/FETCH_HEAD ]; then
+  last_update=`stat -c %Y ${DIR}/.git/FETCH_HEAD`
+  current=`date -d now "+%s"`
+  if [ $(((current - last_update) / 86400)) -gt 1 ]; then
+    echo "Last update ${last_update}, current ${current}: Do git pull spack"
+    cd $DIR && git pull
+  else
+    echo "git pull was less than one day ago"
+  fi
+else
+  echo "git clone spack"
+  git clone https://github.com/spack/spack $DIR || true
+  cd $DIR && git status
+fi
+
+echo "Load spack environment"
+source $DIR/share/spack/setup-env.sh
+
+# Show the known envs
+spack env list | grep ${RUNNER_ENV}
```

Comment:
Error is here because grep doesn't find the environment string (hasn't been created yet).

Reply:
I'm not sure I agree with this. The script executes in the environment generated by the yaml file, and the RUNNER_ENV environment variable is setup in that file.

Reply:
Yes, but if you do "command | grep asdf" and "asdf" is not found, grep will return non-zero.  This causes the shell to exit in this case.

```EXIT STATUS
       Normally,  the  exit  status is 0 if selected lines are found and 1 otherwise.  But the exit status is 2 if an
       error occurred, unless the -q or --quiet or --silent option is used and  a  selected  line  is  found.   Note,
       however,  that  POSIX only mandates, for programs such as grep, cmp, and diff, that the exit status in case of
       error be greater than 1; it is therefore advisable, for the sake of portability, to use logic that  tests  for
       this general condition instead of strict equality with 2.
```